backpack/core/derivatives/shape_check.py

```diff
@@ -185,44 +243,64 @@ def wrapped_mat_prod_check_shapes(self, module, g_inp, g_out, mat, *args, **kwar
 )
 
 residual_mat_prod_accept_vectors = functools.partial(
-    mat_prod_accept_vectors,
+    _mat_prod_accept_vectors,
     vec_criterion=same_dim_as_input,
 )
 
 
 # TODO Refactor using partials
-def make_hessian_mat_prod_accept_vectors(make_hessian_mat_prod):
+def make_hessian_mat_prod_accept_vectors(
+    make_hessian_mat_prod: Callable,
+) -> Callable[..., Callable[..., Tensor]]:
+    """Accept vectors for hessian_mat_prod.
+
+    Args:
+        make_hessian_mat_prod: .
+
+    Returns:
+        .
```

Comment:
Wrapped hessian_mat_prod which converts vector-format inputs to a matrix before processing. Preserves format of input.